doc/source/data/training_ingest.rst

```diff
@@ -0,0 +1,525 @@
+.. _training_ingest_home:
+
+End-to-end: Data Loading for ML Training
+========================================
+
+Data loading & preprocessing is an important step for ML model training. The process involves loading raw data from storage, processing it to the appropriate format, sharding it among training workers, and moving each shard to GPU for training.   
+
+:ref:`Ray Data <data>` works with :ref:`Ray Train <train-docs>` to offer an efficient solution to load and preprocess data for distributed training in a streaming fashion, scaling to very large datasets and avoiding GPU starvation.
+
+For an overview on why you should use Ray Data to ingest data for training, and how it compares to alternatives, see the :ref:`Ray Data Overview <data_overview>`.
+
+.. figure:: images/train_ingest.png
+
+.. _ingest_quickstart:
+
+Quickstart
+----------
+To start, install Ray Data and Ray Train:
+
+.. code-block:: bash
+
+    pip install -U "ray[data,train]"
+
+Using Ray Data and Ray Train for distributed training on large datasets involves 4 basic steps:
+
+- **Step 1:** Load your data into a Ray Dataset. Ray Data supports many different data sources and formats. For more details, see :ref:`Loading Data <loading_data>`.
+- **Step 2:** If required, create a function that defines your training logic and periodically checkpoints your model. For more information, see the :ref:`Ray Train user guides <train-userguides>`.
+- **Step 3:** Inside your training function, access the dataset shard for the training worker via :meth:`session.get_dataset_shard() <ray.air.session.get_dataset_shard>`. Iterate over the dataset shard to train your model. For more details on how to iterate over your data, see the :ref:`Iterating over data <iterating-over-data>` user guide.
```

Comment:
We'll also add a new Train docs discussing how to ingest the ray dataset with TorchTrainer (e.g. lightning, transformers).